presto-main/src/test/java/io/prestosql/operator/scalar/TestMathFunctions.java

```diff
@@ -1142,42 +1142,42 @@ public void testGreatest()
+        assertFunction("greatest(TINYINT'5', TINYINT'4', CAST(NULL AS TINYINT), TINYINT'3')", TINYINT, null);
 
         // smallint
         assertFunction("greatest(SMALLINT'1', SMALLINT'2')", SMALLINT, (short) 2);
         assertFunction("greatest(SMALLINT'-1', SMALLINT'-2')", SMALLINT, (short) -1);
         assertFunction("greatest(SMALLINT'5', SMALLINT'4', SMALLINT'3', SMALLINT'2', SMALLINT'1', SMALLINT'2', SMALLINT'3', SMALLINT'4', SMALLINT'1', SMALLINT'5')", SMALLINT, (short) 5);
         assertFunction("greatest(SMALLINT'-1')", SMALLINT, (short) -1);
-        assertFunction("greatest(SMALLINT'5', SMALLINT'4', CAST(NULL AS SMALLINT), SMALLINT'3')", SMALLINT, (short) 5);
+        assertFunction("greatest(SMALLINT'5', SMALLINT'4', CAST(NULL AS SMALLINT), SMALLINT'3')", SMALLINT, null);
 
         // integer
         assertFunction("greatest(1, 2)", INTEGER, 2);
         assertFunction("greatest(-1, -2)", INTEGER, -1);
         assertFunction("greatest(5, 4, 3, 2, 1, 2, 3, 4, 1, 5)", INTEGER, 5);
         assertFunction("greatest(-1)", INTEGER, -1);
-        assertFunction("greatest(5, 4, CAST(NULL AS INTEGER), 3)", INTEGER, 5);
+        assertFunction("greatest(5, 4, CAST(NULL AS INTEGER), 3)", INTEGER, null);
 
         // bigint
         assertFunction("greatest(10000000000, 20000000000)", BIGINT, 20000000000L);
         assertFunction("greatest(-10000000000, -20000000000)", BIGINT, -10000000000L);
         assertFunction("greatest(5000000000, 4, 3, 2, 1000000000, 2, 3, 4, 1, 5000000000)", BIGINT, 5000000000L);
         assertFunction("greatest(-10000000000)", BIGINT, -10000000000L);
-        assertFunction("greatest(5000000000, 4000000000, CAST(NULL as BIGINT), 3000000000)", BIGINT, 5000000000L);
+        assertFunction("greatest(5000000000, 4000000000, CAST(NULL as BIGINT), 3000000000)", BIGINT, null);
 
         // double
         assertFunction("greatest(1.5E0, 2.3E0)", DOUBLE, 2.3);
         assertFunction("greatest(-1.5E0, -2.3E0)", DOUBLE, -1.5);
         assertFunction("greatest(-1.5E0, -2.3E0, -5/3)", DOUBLE, -1.0);
         assertFunction("greatest(1.5E0, -1.0E0 / 0.0E0, 1.0E0 / 0.0E0)", DOUBLE, Double.POSITIVE_INFINITY);
-        assertFunction("greatest(5, 4, CAST(NULL as DOUBLE), 3)", DOUBLE, 5.0);
+        assertFunction("greatest(5, 4, CAST(NULL as DOUBLE), 3)", DOUBLE, null);
 
         // float
         assertFunction("greatest(REAL '1.5', 2.3E0)", DOUBLE, 2.3);
         assertFunction("greatest(REAL '-1.5', -2.3E0)", DOUBLE, (double) -1.5f);
         assertFunction("greatest(-1.5E0, REAL '-2.3', -5/3)", DOUBLE, -1.0);
         assertFunction("greatest(REAL '1.5', REAL '-1.0' / 0.0E0, 1.0E0 / REAL '0.0')", DOUBLE, (double) (1.0f / 0.0f));
-        assertFunction("greatest(5, REAL '4', CAST(NULL as DOUBLE), 3)", DOUBLE, 5.0);
+        assertFunction("greatest(5, REAL '4', CAST(NULL as DOUBLE), 3)", DOUBLE, null);
```

Comment:
This is a test case for DOUBLE.
Make it into a test case for REAL:

```suggestion
        assertFunction("greatest(5, REAL '4', CAST(NULL AS real), 3)", REAL, null);
```

Reply:
Actually the cases above are also for DOUBLE, not REAL

Reply:
This remained unanswered, so we don't have a test for NULL with REAL type just yet.